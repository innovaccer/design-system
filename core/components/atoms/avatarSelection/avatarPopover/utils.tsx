import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
