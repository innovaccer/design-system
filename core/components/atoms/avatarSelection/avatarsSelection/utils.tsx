import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
