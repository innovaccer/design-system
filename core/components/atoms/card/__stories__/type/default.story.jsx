import { Card } from '@/index';

// CSF format story
export const defaultCard = () => {
  return (
    <Card>
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Default Card',
  component: Card,
};
