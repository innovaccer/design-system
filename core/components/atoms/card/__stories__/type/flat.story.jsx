import { Card } from '@/index';

// CSF format story
export const Flat = () => {
  return (
    <Card shadow="none">
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Flat',
  component: Card,
};
