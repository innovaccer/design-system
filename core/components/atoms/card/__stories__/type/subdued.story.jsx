import { Card, CardSubdued } from '@/index';

// CSF format story
export const Subdued = () => {
  return (
    <CardSubdued>
      <div className="p-8" />
    </CardSubdued>
  );
};

export default {
  title: 'Components/Card/Type/Subdued',
  component: Card,
  subcomponents: { CardSubdued },
  parameters: {
    docs: {
      docPage: {
        title: 'CardSubdued',
        description: 'A component to highlight section of a card.',
      },
    },
  },
};
