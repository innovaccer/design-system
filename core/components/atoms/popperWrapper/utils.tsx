// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
