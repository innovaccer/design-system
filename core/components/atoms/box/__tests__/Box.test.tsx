import * as React from 'react';
import { render } from '@testing-library/react';
import { Box } from '@/index';
import { BoxProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const display: Array<'inline' | 'inline-block' | 'block' | 'flex' | 'inline-flex' | 'grid' | 'inline-grid'> = [
  'inline',
  'inline-block',
  'block',
  'flex',
  'inline-flex',
  'grid',
  'inline-grid',
];

const shadow: Array<'none' | 's' | 'm' | 'l'> = ['none', 's', 'm', 'l'];

const borderRadius: Array<'none' | '2-5' | '05' | '10' | '15' | '20' | '30' | '40' | 'full'> = [
  'none',
  '2-5',
  '05',
  '10',
  '15',
  '20',
  '30',
  '40',
  'full',
];

const padding = ['none', 'xs', 's', 'm', 'l', 'xl'];

const background: Array<
  'transparent' | 'light' | 'secondary-lightest' | 'primary' | 'secondary' | 'success' | 'warning' | 'danger'
> = ['transparent', 'light', 'secondary-lightest', 'primary', 'secondary', 'success', 'warning', 'danger'];

const borderColor: Array<'none' | 'default' | 'primary' | 'secondary' | 'success' | 'warning' | 'danger'> = [
  'none',
  'default',
  'primary',
  'secondary',
  'success',
  'warning',
  'danger',
];

const position: Array<'static' | 'relative' | 'absolute' | 'fixed' | 'sticky'> = [
  'static',
  'relative',
  'absolute',
  'fixed',
  'sticky',
];

const gap: Array<'xs' | 's' | 'm' | 'l'> = ['xs', 's', 'm', 'l'];
const flex: Array<1 | 2 | 3 | 4> = [1, 2, 3, 4];

describe('Box component', () => {
  const mapper: Record<string, any> = {
    display: valueHelper(display, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    shadow: valueHelper(shadow, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    borderRadius: valueHelper(borderRadius, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    padding: valueHelper(padding, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    background: valueHelper(background, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    borderColor: valueHelper(borderColor, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    position: valueHelper(position, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Box {...attr}>Box Content</Box>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component with custom dimensions', () => {
  it('renders with width and height props', () => {
    const { baseElement } = render(
      <Box width="200px" height="100px">
        Box Content
      </Box>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('renders with percentage values', () => {
    const { baseElement } = render(
      <Box width="50%" height="50%">
        Box Content
      </Box>
    );
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Box component with custom element', () => {
  it('renders with a custom HTML element', () => {
    const { baseElement } = render(<Box as="section">Box Content</Box>);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders with a custom React component', () => {
    const CustomComponent = ({ children, ...props }: any) => <span {...props}>{children}</span>;
    const { baseElement } = render(<Box as={CustomComponent}>Box Content</Box>);
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Box component with className', () => {
  it('applies custom className', () => {
    const { getByTestId } = render(<Box className="custom-class">Box Content</Box>);
    expect(getByTestId('DesignSystem-Box')).toHaveClass('custom-class');
  });
});

describe('Box component with data attributes', () => {
  it('applies HTML attributes', () => {
    const { getByTestId } = render(<Box data-custom="test-value">Box Content</Box>);
    expect(getByTestId('DesignSystem-Box')).toHaveAttribute('data-custom', 'test-value');
  });
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    gap: valueHelper(gap, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Box display="flex" {...attr}>
          Box Content
        </Box>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Box component', () => {
  const mapper: Record<string, any> = {
    flex: valueHelper(flex, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Box display="flex" {...attr}>
          Box Content
        </Box>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
