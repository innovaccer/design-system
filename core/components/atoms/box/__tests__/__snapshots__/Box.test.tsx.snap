// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component 
  background: "danger"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--danger Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "light"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--light Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "primary"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--primary Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "secondary"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--secondary Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "secondary-lightest"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--secondary-lightest Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "success"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--success Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "transparent"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  background: "warning"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--warning Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "danger"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--danger Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "default"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--default Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "none"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "primary"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--primary Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "secondary"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--secondary Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "success"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--success Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderColor: "warning"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--warning Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "05"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--05 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "2-5"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--2-5 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "10"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--10 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "15"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--15 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "20"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--20 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "30"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--30 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "40"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--40 Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "full"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--full Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  borderRadius: "none"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "block"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "flex"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "grid"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--grid Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "inline"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--inline Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "inline-block"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--inline-block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "inline-flex"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--inline-flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  display: "inline-grid"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--inline-grid Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  flex: 1
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-flex--1"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  flex: 2
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-flex--2"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  flex: 3
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-flex--3"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  flex: 4
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-flex--4"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  gap: "l"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-gap--l"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  gap: "m"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-gap--m"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  gap: "s"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-gap--s"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  gap: "xs"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--flex Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static Box-gap--xs"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "l"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="l"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "m"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="m"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "none"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="none"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "s"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="s"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "xl"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="xl"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  padding: "xs"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      padding="xs"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  position: "absolute"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--absolute"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  position: "fixed"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--fixed"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  position: "relative"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--relative"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  position: "static"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  position: "sticky"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--sticky"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  shadow: "l"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--l Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  shadow: "m"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--m Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  shadow: "none"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component 
  shadow: "s"
 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--s Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component with custom dimensions renders with percentage values 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      style="width: 50%; height: 50%;"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component with custom dimensions renders with width and height props 1`] = `
<body>
  <div>
    <div
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
      style="width: 200px; height: 100px;"
    >
      Box Content
    </div>
  </div>
</body>
`;

exports[`Box component with custom element renders with a custom HTML element 1`] = `
<body>
  <div>
    <section
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </section>
  </div>
</body>
`;

exports[`Box component with custom element renders with a custom React component 1`] = `
<body>
  <div>
    <span
      class="Box Box-display--block Box-shadow--none Box-borderRadius--none Box-background--transparent Box-borderColor--none Box-position--static"
      data-test="DesignSystem-Box"
    >
      Box Content
    </span>
  </div>
</body>
`;
